network-api/networkapi/wagtailpages/tests/buyersguide/test_forms.py

```diff
@@ -0,0 +1,117 @@
+from django.test import TestCase
+from wagtail.admin.edit_handlers import get_form_for_model
+from wagtail.tests.utils.form_data import nested_form_data, rich_text, streamfield
+
+from networkapi.wagtailpages import models as pagemodels
+from networkapi.wagtailpages.factory import buyersguide as buyersguide_factories
+from networkapi.wagtailpages.pagemodels.buyersguide.forms import (
+    BuyersGuideArticlePageForm,
+)
+
+
+class BuyersGuideArticlePageFormTest(TestCase):
+    def setUp(self):
+        self.article_page_form = get_form_for_model(
+            model=pagemodels.BuyersGuideArticlePage, form_class=BuyersGuideArticlePageForm
+        )
+
+    @staticmethod
+    def generate_form_data(data: dict) -> dict:
+        """
+        Generate valid form data for the BuyersGuide article page form.
+
+        We add the extra fields below to the dict as all these fields are set as
+        required, and the form will return a validation error against them if left out.
+
+        Since we have the "body" field set as required, we need to submit at least
+        one StreamField in the form, or else the form will return a validation error.
+        """
+        return nested_form_data(
+            {
+                **data,
+                "title": "Test Article Title",
+                "slug": "test-article-title",
+                "body": streamfield([("paragraph", rich_text("Hello world!"))]),
+            }
+        )
+
+    def test_article_page_requires_search_image(self):
+        """
+        Test that the buyersguide article page form will
+        return a validation error if no "search_image" is set.
+        """
+        test_form = self.article_page_form(
+            data=self.generate_form_data(
+                {
+                    "search_description": "Test search description",
+                    "search_image": None,
+                },
+            ),
+        )
+
+        self.assertFalse(test_form.is_valid())
+        self.assertEqual(1, len(test_form.errors))
+        self.assertIn("search_image", test_form.errors)
+        self.assertIn("This field is required.", test_form.errors["search_image"])
+
+    def test_article_page_requires_search_description(self):
+        """
+        Test that the buyersguide article page form will
+        return a validation error if no "search_description" is set.
+        """
+        article_page = buyersguide_factories.BuyersGuideArticlePageFactory()
```

Comment:
Can you add some context why we still need to generate the article page if we don't set the `instance` anymore? If this is just to have a convenient way to pass a valid image id, what do you think about using `networkapi.wagtailpages.factory.image_factory.ImageFactory` and passing `ImageFactory().id`?

Reply:
Good question and great suggestion Tibor! I have updated the tests to no longer use the article page factory and instead use the image factory when we need it. I think this makes things simpler as we have a set string for the search description tests, and only use a factory for the image. Updated!